Improve stacking performance and footprint

kurkle opened the pull request:
# Performance
## Master

![image](https://user-images.githubusercontent.com/27971921/69036081-c37efd00-09ed-11ea-8be0-e4190c9ea224.png)

## PR

![image](https://user-images.githubusercontent.com/27971921/69036113-de517180-09ed-11ea-8b96-d622cb991cd3.png)

About ~200ms improvement overall in uPlot stacked (~950ms -> ~750ms).

# Memory

## Master

![image](https://user-images.githubusercontent.com/27971921/69036541-cc240300-09ee-11ea-86c7-5e2988e6eb6d.png)

## PR

![image](https://user-images.githubusercontent.com/27971921/69036500-b7476f80-09ee-11ea-964d-290b46dd66f1.png)

benmccann commented on src/core/core.datasetController.js:
```diff
@@ -257,12 +296,14 @@ helpers.extend(DatasetController.prototype, {
+		const dataset = me.getDataset();
+		const xid = meta.xAxisID = dataset.xAxisID || getFirstScaleId(chart, 'x');
+		const yid = meta.yAxisID = dataset.yAxisID || getFirstScaleId(chart, 'y');
+		meta.xScale = me.getScaleForId(xid);
```
this is a really good idea!

benmccann replied:
do you think we should store `iScale` and `vScale` here as well?

kurkle replied:
Probably

benmccann commented on src/controllers/controller.line.js:
```diff
@@ -106,15 +103,15 @@ module.exports = DatasetController.extend({
 	updateElement: function(point, index, reset) {
-		var me = this;
-		var meta = me.getMeta();
-		var xScale = me._xScale;
```
do we still need to set `_xScale` since it's not used here?

kurkle replied:
Sorry, I'm not sure what you mean?

benmccann replied:
Nevermind. You already addressed it

kurkle commented:
Forgot to mention the leading idea: in `_parsed`, store reference to the stack object instead of the key that leads to it.

kurkle commented:
Added vScale / iScale. Also replaced getMeta() with _cachedMeta. Updated bunch of `var` to `const` / `let`. Not by all means complete on those `var`'s. I can remove those, if its too much to review.
